[#68834] Fix Time Entry tooltips in My spent time

myabc opened the pull request:
# Ticket

https://community.openproject.org/wp/68834

# What are you trying to accomplish?

Implements tooltips using a Primer-style Popover - a `FIXME` from the jQuery removal work.

## Screenshots

<img width="783" height="716" alt="Screenshot 2025-11-10 at 15 30 01" src="https://github.com/user-attachments/assets/bab97540-97a2-4830-8e2c-1bc6e1540d0d" />

# What approach did you choose and why?

~~This (ab)uses Primer's tooltip - as it shows multiple lines of text (actually a list). Tooltips generally shouldn't contain this much information, but this was the quickest fix (and hopefully takes care of accessibility).~~

# Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [ ] Tested major browsers (Chrome, Firefox, Edge, ...)

myabc committed:
[#68834] Fix Time Entry tooltips in My spent time

Implements tooltips using Primer's `tool-tip` custom element.

myabc commented:
@HDinger I think we can probably delete some of the styles in `src/global_styles/content/_tooltip.sass` as we're no longer using jQuery UI tooltip (AFAICT Backlogs doesn't use it)

myabc committed:
Fix missing closing tag in tooltip <ul>

myabc commented on frontend/src/app/features/calendar/te-calendar/te-calendar.component.ts:
```diff
@@ -534,23 +535,28 @@ export class TimeEntryCalendarComponent implements AfterViewInit, OnDestroy {
+    ensureId(tooltip, 'tooltip');
+    tooltip.setAttribute('for', forId);
+    tooltip.setAttribute('popover', 'manual');
+    tooltip.setAttribute('data-direction', 'sw');
```
Is this the right direction?

myabc replied:
Changed direction to the default, 's'

mrmir replied:
I think 's' works quite well, but maybe @HDinger thinks otherwise?

HDinger replied:
I don't have a strong preference on this. Let's go with `s` for now. We can still change that later on.

myabc commented on frontend/src/app/features/calendar/te-calendar/te-calendar.component.ts:
```diff
@@ -534,23 +535,28 @@ export class TimeEntryCalendarComponent implements AfterViewInit, OnDestroy {
+    tooltip.setAttribute('for', forId);
+    tooltip.setAttribute('popover', 'manual');
+    tooltip.setAttribute('data-direction', 'sw');
+    tooltip.setAttribute('data-type', 'label');
```
Actually, `aria-describedby` probably makes more sense here.

myabc commented on frontend/src/app/features/calendar/te-calendar/te-calendar.component.ts:
```diff
@@ -534,23 +535,28 @@ export class TimeEntryCalendarComponent implements AfterViewInit, OnDestroy {
+    const forEl = event.el;
+    const forId = ensureId(forEl);
+
     const tooltip = document.createElement('tool-tip');
```
We're not importing the Primer typings, which is annoying.

myabc committed:
Remove styles for no longer used jQuery UI tooltip

mrmir reviewed (approved):
Works well 👍🏽

mrmir reviewed (changes requested):
For the bug we discussed already regarding the tooltip not moving with the cursor

myabc committed:
Rework Time Entry tooltips as Popovers

Uses Primer's Popover component styles along with `anchored-position`
custom element to position the popover.

Switches to Primer utility classes for styling.

See https://primer.style/product/components/popover/
See https://github.com/primer/behaviors/blob/main/src/anchored-position.ts

myabc committed:
Add lit-html dependency for HTML templating

myabc committed:
Use lit-html templates to tidy up Popover creation

myabc committed:
eslint fixes: formatting, nullish operator

myabc committed:
Fix unnecessary async without await

mrmir reviewed (approved):
Looks fine as a popover, but the tracking/following behaviour needs to be figured out
